docs/source/function_overloading.rst

```diff
@@ -65,6 +65,24 @@ accurately describe the function's behavior.
             else:
                 raise TypeError(...)
 
+Calls to overloaded functions are type checked against the variants,
+not against the implementation. A call like ``my_list[5]`` would have
+type ``T``, not ``Union[T, Sequence[T]]`` because it matches the first
+overloaded definition, and ignores the type annotations on the
+implementation of ``__getitem__``. The code in the body of the
+definition of ``__getitem__`` is checked against the annotations on
+the last function declaration. In this case the body is checked with
+``index: Union[int, slice]`` and a return type
+``Union[T, Sequence[T]]``. If there are no annotations on the last
+definition, then code in the function body is not type checked. The
+annotations on the function body must be compatible with the types
+given for the overloaded variants listed above it. The type checker
+will ensure that all the types listed for parameters and the return
```

Comment:
ensure -> verify

(It can't ensure anything, it has no powers except for pointing out errors. :-)